lib/ClangImporter/ImportType.cpp

```diff
@@ -1986,9 +1979,11 @@ Type ClangImporter::Implementation::importMethodType(
         swiftResultTy = OptionalType::get(OptionalityOfReturn, swiftResultTy);
     }
   }
+
   if (!swiftResultTy)
     return Type();
-  swiftResultTy = mapTypeIntoContext(swiftResultTy);
+
+  swiftResultTy = mapTypeIntoContext(origDC, dc, swiftResultTy);
```

Comment:
@DougGregor The reason why I left that in place is because I noticed in case `fromDC != toDC` if I simply do `toDC->mapTypeIntoContext(type)` it always produces `ErrorType`